understanding/21/non-text-contrast.html

```diff
@@ -66,22 +66,23 @@ <h2>Intent</h2>
+				<p>For active controls on the page, such as buttons and form fields, any visual information provided that is necessary for a user to identify that a control exists and how to operate it must have sufficient contrast with the adjacent background. Also, the visual effects which are implemented to order to indicate state, such as whether a component is selected or focused, must also ensure that the information used to identify the control in that state has a minimum 3:1 contrast ratio.</p>
 				<figure>
 					<!-- example from https://getbootstrap.com/docs/4.1/components/buttons/ -->
 					<img alt="Three buttons on a white background. The first has a dark grey border to indicate its boundary, white internal background and black text. The second adds a thick grey outer border to show focus. The third has the grey border with black background and white text." src="img/first-button-example.png" /> 
 					<figcaption>An active control showing the visual boundary by default, the focus indicator, and the hover state.</figcaption>
 				</figure>
 				
-				<p>This success criteria does not require that controls have a visual boundary indicating the hit area, but that if there is an indicator then it has sufficient contrast. If only the text (or icon) is visible and there is no visual indication of the hit area, then there is no contrast requirement beyond the text contrast (<a href="visual-audio-contrast-contrast.html">1.4.3 Contrast (Minimum)</a>) or icon contrast covered by the Graphical Objects portion of this Success Criteria. Note that for people with cognitive disabilities it is recommended to delineate the boundary of controls to aid in the recognition of controls and therefore the  completion of activities.</p>
-				<p>Regardless of the whether or not there is a visible boundary for the button, the focus indicator for the component must have sufficient contrast when the component is focused.</p>
+				<p>This success criteria does not require that controls have a visual boundary indicating the hit area, but that if the visual boundary is the only way to identify that a control can be activated, then that boundary must have sufficient contrast. If text (or an icon) within a button is visible and there is no visual indication of the hit area, or the visual indication does not provide the only indication, then there is no contrast requirement beyond the text contrast (<a href="visual-audio-contrast-contrast.html">1.4.3 Contrast (Minimum)</a>) or icon contrast covered by the Graphical Objects portion of this Success Criteria. Note that for people with cognitive disabilities it is recommended to delineate the boundary of controls to aid in the recognition of controls and therefore the completion of activities.</p>
+        
+				<p>In all cases, the visual focus indicator for a component must have sufficient contrast against the adjacent background when the component is focused.</p>
```

Comment:
The SC text adds
> except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author;

Reply:
Thanks David, made edits that I think cover it but a bit shorter.